### DISCUSSION TITLE
Overlap of peaks called in each sample of merged dataset

### DISCUSSION THREAD

**Discussion by sneddonucsf** [GreenleafLab/ArchR#1736] [Questions / Documentation] [answered]
Hello, 

I have two samples in a merged dataset that I have then called peaks on. I was wondering how it might be possible to compare the overlap of called peaks in for each sample; i.e. see how reproducible called peaks are in each cluster on a per-sample basis?

**Comment by rcorces** [comment, marked as answer]
I typically discourage direct comparison of called peaks as a means of understanding biology because I find peak calling to have many confounders but perhaps thats not how you plan to interpret this. 
We dont provide methods to do this but standard GRanges operations should work and the per-sample peak calls are stored within the `./PeakCalls` directory. Does that answer your question?

**Reply by sneddonucsf** [reply]
@rcorces thank you for your reply! My intended use is to look at called peaks in each of my samples within a single cluster of a merged ArchR project and see the overlap/reproducibility between the two (i.e. within cluster 1, compared peaks called in sample 1 vs. sample 2). This is to answer a reviewer comment where they wanted us to do a comparison of called peaks in each sample within a given cluster to assess robustness/reproducibility between the two samples. 

Is this possible with the reproducible peaks, or is that sample-specific information lost during the iterative overlap process? I've tried wrapping my head around the process with the manual, but I'm still a little lost.

**Reply by rcorces** [reply]
Peak calls in ArchR are done on pseudo-bulk replicates. If your pseudo-bulk replicates are created in such a way that they each only represent a single sample, then the peak call files stored in the PeakCalls sub directories would be what you are looking for. Otherwise, you could export per-sample per-cluster fragments using [`getFragmentsFromProject()`](https://www.archrproject.com/reference/getFragmentsFromProject.html) and then call peaks outside of ArchR using macs2.

**Reply by sneddonucsf** [reply]
Thanks for the answer @rcorces. I was a little confused when trying to use Macs2 externally, so I instead opted to subset my ArchR project by the cluster I want to compare, then further subsetted into two separate ArchR projects for each sample, called peaks on each of those individually with ```addGroupCoverages()``` and ```addReproduciblePeakSet()``` and grabbed the peaks with ```getPeakSet()``` and looked at the overlap via GRanges. Does this sound like an acceptable workflow?

**Reply by rcorces** [reply]
Its hard for me to wrap my head around how this would all work and I dont like to make comments on specific analyses. Beyond the recommendations given above, I dont have much to add.

**Reply by sneddonucsf** [reply]
All good, thanks @rcorces!
